# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PContent.ipdl
# Commit: a008d3cebfdb
# Full Hash: a008d3cebfdb9d96ab32530b90180141575316a8
# Author: alwu <alwu@mozilla.com>
# Date: 2020-03-17 09:36:40
# Regressor Bug: 1582508
# File Overlap Count: 1
# Description:
#   Bug 1582508 - part4 : propagate declared playback state to media session controller. r=chunmin
#   
#   This patch includes the implementation of propagating declared playback state from the media session in the content process to the media session controller in the chrome process.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D66341
# ==============================================================================

diff -r a73415e9c006 -r a008d3cebfdb dom/ipc/PContent.ipdl
--- a/dom/ipc/PContent.ipdl	Tue Mar 17 00:03:44 2020 +0000
+++ b/dom/ipc/PContent.ipdl	Tue Mar 17 00:17:36 2020 +0000
@@ -119,6 +119,7 @@
 using mozilla::dom::MediaControlKeysEvent from "ipc/MediaControlIPC.h";
 using mozilla::dom::ControlledMediaState from "ipc/MediaControlIPC.h";
 using mozilla::dom::MaybeMediaMetadataBase from "mozilla/dom/MediaSessionIPCUtils.h";
+using mozilla::dom::MediaSessionPlaybackState from "mozilla/dom/MediaSessionBinding.h";
 using refcounted class nsDocShellLoadState from "nsDocShellLoadState.h";
 using mozilla::dom::ServiceWorkerShutdownState::Progress from "mozilla/dom/ServiceWorkerShutdownState.h";
 using refcounted class mozilla::dom::CrossProcessSHEntry from "mozilla/dom/MaybeNewPSHEntry.h";
@@ -1500,6 +1501,14 @@
                                     MaybeMediaMetadataBase aMetadata);
 
     /**
+     * This method is used to update media session's playback state whenever its
+     * playback state is changed.
+     */
+    async NotifyMediaSessionPlaybackStateChanged(
+        MaybeDiscardedBrowsingContext aContext,
+        MediaSessionPlaybackState aMetadata);
+
+    /**
      * Due to sandboxing, a child process's UntrustedModulesProcessor cannot
      * obtain enough information about a DLL file to determine its
      * trustworthiness. This API asks the chrome process to perform that